sdk/identity/identity/src/credentials/deviceCodeCredential.ts

```diff
@@ -89,138 +74,18 @@ export class DeviceCodeCredential implements TokenCredential {
-          `Failed to authenticate ${(err as AuthenticationError).errorResponse.errorDescription}`
-        );
-      } else {
-        logger.info(`Failed to authenticate ${err}`);
-      }
-
-      span.setStatus({
-        code,
-        message: err.message
-      });
-      throw err;
-    } finally {
-      span.end();
-    }
-  }
-
-  private async pollForToken(
-    deviceCodeResponse: DeviceCodeResponse,
-    options?: GetTokenOptions
-  ): Promise<TokenResponse | null> {
-    let tokenResponse: TokenResponse | null = null;
-    const { span, options: newOptions } = createSpan("DeviceCodeCredential-pollForToken", options);
-
-    try {
-      const webResource = this.identityClient.createWebResource({
-        url: `${this.identityClient.authorityHost}/${this.tenantId}/oauth2/v2.0/token`,
-        method: "POST",
-        disableJsonStringifyOnBody: true,
-        deserializationMapper: undefined,
-        body: qs.stringify({
-          grant_type: "urn:ietf:params:oauth:grant-type:device_code",
-          client_id: this.clientId,
-          device_code: deviceCodeResponse.device_code
-        }),
-        headers: {
-          Accept: "application/json",
-          "Content-Type": "application/x-www-form-urlencoded"
-        },
-        abortSignal: options && options.abortSignal,
-        spanOptions: newOptions.tracingOptions && newOptions.tracingOptions.spanOptions
-      });
-
-      while (tokenResponse === null) {
-        try {
-          // Referencing delay from core-http this way for testing purposes.
-          await coreHttp.delay(deviceCodeResponse.interval * 1000);
 
-          // Check the abort signal before sending the request
-          if (options && options.abortSignal && options.abortSignal.aborted) {
-            return null;
-          }
-
-          tokenResponse = await this.identityClient.sendTokenRequest(webResource);
-        } catch (err) {
-          if (err.name === AuthenticationErrorName) {
-            switch (err.errorResponse.error) {
-              case "authorization_pending":
-                break;
-              case "authorization_declined":
-                return null;
-              case "expired_token":
-                throw err;
-              case "bad_verification_code":
-                throw err;
-              default:
-                // Any other error should be rethrown
-                throw err;
-            }
-          } else {
-            throw err;
-          }
-        }
-      }
-
-      return tokenResponse;
-    } catch (err) {
-      const code =
-        err.name === AuthenticationErrorName
-          ? CanonicalCode.UNAUTHENTICATED
-          : CanonicalCode.UNKNOWN;
-      span.setStatus({
-        code,
-        message: err.message
-      });
-      throw err;
-    } finally {
-      span.end();
-    }
+    const publicClientConfig = {
+      auth: {
+          clientId: this.clientId,
+          authority: "https://login.microsoftonline.com/" + this.tenantId,
```

Comment:
This should be configurable as with the `InteractiveBrowserrCredential`

Reply:
Missed that one. Updated